modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+      <productId>111</productId>
+      <price>8.90</price>
+    </i_2>
+  </itemList>
+  <itemList>
+    <i_3>
+      <id>31</id>
+      <quantity>7</quantity>
+      <productId>222</productId>
+      <price>5.20</price>
+    </i_3>
+  </itemList>
+  <itemList>
+    <i_4>
+      <id>31</id>
+      <quantity>7</quantity>
+      <productId>222</productId>
+      <price>5.20</price>
+    </i_4>
+  </itemList>
+</salesorder>
+----
+​
+== Validate that a Script is Streamable (Experimental Feature)
+​
+To help you code in a streamable way and avoid mistakes that can break the streaming process, DataWeave provides an _advanced, experimental_ annotation and a related directive:
+
+* `@StreamCapable()` +
+annotation: Use this annotation to validate whether the script can sequentially access a variable (typically the `payload` variable).
+* `input` directive: +
+The `@StreamCapable()` annotation requires the use of an input directive in the DataWeave script that identifies the MIME type of the data source, for example, `input payload application/xml`.
+
+The validator checks a script against the following criteria:
+
+* The variable is referenced only once
+* No index selector is set for negative access, such as `[-1]`
+* No reference to the variable is found in a nested lambda
+
+If all are true, the selected data is streamable. The following example validates successfully. The script is designed to act on the <<example_json, JSON input>> from the <<stream_json, JSON streaming>> section.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+
+@StreamCapable()
+input payload application/json streaming=true
+output application/json
+---
+payload.family filter (member) -> member.age > 3
+----
+
+The script successfully validates and returns the following output:
+
+[source,JSON,linenums]
+----
+[
+  {
+    "name": "Pedro",
+    "age": 4
+  },
+  {
+    "name": "Matias",
+    "age": 8
+  }
+]
+----
+
+=== Validation Failures
+
+If any of the criteria that the valiator checks are false, the validation fails. 
+
+Before proceeding, note that validation can fail in some cases when streaming works. If you write a script in a way that sequentially accesses the input variable in a given data source, streaming works, but that script might not work in all cases. For example, JSON does not place a restriction on the order of the keys in an object. If the keys in some JSON documents arrive in a different order than the script expects, streaming will fail in those cases. The annotation processor follows the rules of the format and cannot assume that the keys always arrive in the same order.
+
+==== Error: Variable is Referenced More Than Once
+
+Validation fails if a script attempts to reference the same variable more than once.
+
+The following script is designed to act on the <<example_json, JSON input>> from the <<stream_json, JSON streaming>> section. Validation fails because the script attempts to reference the `payload` variable more than once.
```

Comment:
```suggestion
The following script is designed to act on the <<example_json, JSON input>> from the <<stream_json, JSON streaming>> section. Validation fails because the script attempts to reference the `payload` variable more than once:
```